src/Source/CopcSource.js

```diff
@@ -0,0 +1,121 @@
+import Source from 'Source/Source';
+import * as THREE from 'three';
+
+/**
+ * @param {function(number, number):Promise<Uint8Array>} fetcher
+ */
+async function getHeaders(fetcher) {
+    const header =
+        Las.Header.parse(await fetcher(0, Las.Constants.minHeaderLength));
+    const vlrs = await Las.Vlr.walk(fetcher, header);
+
+    // info VLR: required by COPC
+    const infoVlr = Las.Vlr.find(vlrs, 'copc', 1);
+    if (!infoVlr) { return Promise.reject('COPC info VLR is required'); }
+    const info = Info.parse(await Las.Vlr.fetch(fetcher, infoVlr));
+
+    // OGC Coordinate System WKT: required by LAS1.4
+    const wktVlr = Las.Vlr.find(vlrs, 'LASF_Projection', 2112);
+    if (!wktVlr) { return Promise.reject('LAS1.4 WKT VLR is required'); }
+    const wkt = Binary.toCString(await Las.Vlr.fetch(fetcher, wktVlr));
+
+    // Extra bytes: optional by LAS1.4
+    const ebVlr = Las.Vlr.find(vlrs, 'LASF_Spec', 4);
+    const eb = ebVlr ?
+        Las.ExtraBytes.parse(await Las.Vlr.fetch(fetcher, ebVlr)) :
+        [];
+
+    return { header, info, wkt, eb };
+}
+
+/**
+ * @classdesc
+ * A source for [Cloud Optimised Point Cloud](https://copc.io/) (COPC) data.
+ * Such data consists of a [LAZ 1.4](https://www.ogc.org/standard/las/) file
+ * that stores compressed points data organized in a clustered octre.
+ *
+ * A freshly created source fetches and parses portions of the file
+ * corresponding to the LAS 1.4 header, all the Variable Length Record (VLR)
+ * headers as well the following VLRs:
+ * - COPC [`info`](https://copc.io/#info-vlr) record (mandatory)
+ * - LAS 1.4 `OGC Coordinate System WKT` record (mandatory, see [Las 1.4
+ *   spec](https://portal.ogc.org/files/?artifact_id=74523))
+ * - LAS 1.4 `Extra Bytes` record (optional, see [Las 1.4
+ *   spec](https://portal.ogc.org/files/?artifact_id=74523))
+ *
+ * @extends {Source}
+ *
+ * @property {boolean} isCopcSource - Read-only flag to check that a given
+ * object is of type CopcSource.
+ * @property {Object} header - LAS header of the source.
+ * @property {Object[]} eb - List of headers of each Variable Length Records
+ * (VLRs).
+ * @property {Object} info - COPC `info` VLR.
+ * @property {number[]} info.cube - Bouding box of the octree as a 6-elements.
+ * tuple `[minX, minY, minZ, maxX, maxY, maxZ]`. Computed from `center_x`,
+ * `center_y`, `center_z` and `halfSize` properties.
+ * @property {Object} info.rootHierarchyPage - Hierarchy page of the root node.
+ * @property {number} info.rootHierarchyPage.pageOffset - Absolute Offset to the
+ * root node data chunk.
+ * @property {number} info.rootHierarchyPage.pageOffset - Size (in bytes) of the
+ * root node data chunk.
+ * @property {number[]} gpsTimeRange - A 2-element tuple denoting the minimum
+ * and maximum values of attribute `gpsTime`.
+ */
+class CopcSource extends Source {
+    /**
+     * @param {Object} config - Source configuration
+     * @param {string} config.url - URL of the COPC ressource.
```

Comment:
```suggestion
     * @param {string} config.url - URL of the COPC resource.
```
Little mistake made multiple times in the comments